### ISSUE TITLE
Can Gaussian Processes be incorporated?

### ISSUE THREAD

**Issue by ibab** [tensorprob/tensorprob#57]
Can we find a way to introduce Gaussian Processes (See [here](https://www.cs.toronto.edu/~hinton/csc2515/notes/gp_slides_fall08.pdf) for an excellent introduction) as a `Distribution` that's available for model building?

I'd love to try them out for modeling background distributions in HEP (if that even makes sense :confused:)

**Comment by cranmer** [comment]
main issue with GPs speed wise is linear algebra.

See HODLR solver and paper implemented in this nice GP package:
https://github.com/dfm/george/blob/master/docs/index.rst

It does make sense, pretty much anywhere you see polynomial or some other ad hoc function, but there are some subtle issues with marginalization vs. profiling in the fits.
Here's an example I'm working on for dijet 
https://github.com/cranmer/play/blob/master/misc/DiJet_GP_test.ipynb

**Comment by ibab** [comment]
Thanks for the links! I will have a look at the notebook.
Maybe we could allow users to "import" a GP from George as a distribution into TensorProb.

**Comment by cranmer** [comment]
Yeah, there is this issue that a GP is really a prior over all functions.
There is a posterior mean that is sort of like the best fit model, but that
happens analytically (no external optimization needed). Some work needed to
translate into a Frequentist/ profile language consistently

Kyle

**Comment by cranmer** [comment]
Also, to be a PDF you can do a GP on logP(X) -- a log Gaussian process --
to I force positivity and then normalize (or absorb integral into Poisson
part of extended PDF). That gets a little wonky at times.

Including the Poisson part fluctuations on top makes a double stochastic
process called a Cox process, but that's a mess numerically
